.github/workflows/ci.yml

```
@@ -74,7 +74,7 @@ jobs:
     - name: Install Rust Nightly
       uses: actions-rs/toolchain@v1
       with:
-        toolchain: nightly-2021-01-01
+        toolchain: nightly-2021-11-01
```

Let's go ahead and bump this a little further; this is mostly arbitrary.

```suggestion
        toolchain: nightly-2022-01-01
```

My current understanding of best practice is different from what we're currently doing here, though. *Probably in a follow-up*, it would be better practice to have

- check
  - +pin-nightly generate-lockfile -Zminimal-versions
  - +pin bootstrap --locked
  - +msrv check --locked
- lint
  - +pin-nightly generate-lockfile -Zminimal-versions
  - +pin bootstrap --locked
  - +pin clippy --locked -Dwarnings
  - +pin fmt --check
- test
  - +pin-nightly generate-lockfile -Zminimal-versions
  - +pin bootstrap --locked
  - +pin-stable test --locked
- future (allow-fail)
  - +beta generate-lockfile
  - +beta bootstrap --locked
  - +beta test --locked

The goal is to avoid anything changing between CI runs except the local code.